one/src/events.rs

```diff
@@ -156,3 +376,215 @@ async fn migrate_from_database(input_ceramic_db: PathBuf, store: SQLiteBlockStor
     );
     result.unwrap()
 }
+
+#[derive(Debug, Clone)]
+pub struct SQLiteRootStore {
+    pool: SqlitePool,
+}
+
+impl SQLiteRootStore {
+    // ```sql
+    // CREATE TABLE recon (root BLOB, timestamp INTEGER)
+    // ```
+    pub async fn new(pool: SqlitePool) -> Result<Self> {
+        let store = Self { pool };
+        store.create_table_if_not_exists().await?;
+        Ok(store)
+    }
+
+    async fn create_table_if_not_exists(&self) -> Result<()> {
+        // this will need to be moved to migration logic if we ever change the schema
+        sqlx::query("CREATE TABLE IF NOT EXISTS roots (root BLOB, timestamp INTEGER, PRIMARY KEY(root, timestamp));",
+        )
+        .execute(self.pool.writer())
+        .await?;
+        Ok(())
+    }
+
+    /// Store root, timestamp.
+    pub async fn put(&self, root: &[u8], timestamp: i64) -> Result<()> {
+        match sqlx::query("INSERT OR IGNORE INTO roots (root, timestamp) VALUES (?, ?)")
+            .bind(root)
+            .bind(timestamp)
+            .execute(self.pool.writer())
+            .await
+        {
+            Ok(_) => Ok(()),
+            Err(err) => Err(err.into()),
+        }
+    }
+
+    pub async fn get(&self, root: &[u8]) -> Result<Option<i64>> {
+        Ok(sqlx::query("SELECT timestamp FROM roots WHERE root = ?;")
+            .bind(root)
+            .fetch_optional(self.pool.reader())
+            .await?
+            .map(|row| row.get::<'_, i64, _>(0)))
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
are there more tests in js-ceramic we should add? it seems like we should have at least an invalid time event test